Detectors/EMCAL/simulation/src/DMAOutputStream.cxx

```diff
@@ -30,7 +30,19 @@ void DMAOutputStream::open()
   }
 }
 
-void DMAOutputStream::writeData(RawHeader header, gsl::span<char> buffer)
+void DMAOutputStream::writeSingleHeader(RawHeader header) 
```

Comment:
Should not you have here  ``const RawHeader& header`` ?  I see I modify this header, but it looks like a bug, see below.

Reply:
I was under the impression that the header format is so small that there is no significant copy overhead.  In this function the header is not modified, but in the one above the header must be modified in order to adapt the page counter, so a copy is necessary in any case.

Reply:
I don't understand this: you are anyway creating a local copy of the stack object ``header`` in the ``outputheader``.
Then you keep modifying the ``header`` on the stack (the modification will be lost on return) although it is not used anymore. I think you should modify the ``outputheader`` instead, see my comment below.

Reply:
Right, this is a bug, sorry for overlooking. Will be fixed.